backend/cmm.mli

```diff
@@ -178,18 +178,28 @@ type memory_chunk =
   | Onetwentyeight_unaligned           (* word-aligned 128-bit vector *)
   | Onetwentyeight_aligned             (* 16-byte-aligned 128-bit vector *)
 
-type vector_cast =
-  | Bits128
-
-type scalar_cast =
-  (* CR mslater: move all bit-casts into a reinterpret_cast type *)
-  | Float32_as_float
-  | Float_to_int of float_width
+(* These casts compile to a single move instruction. If the operands are assigned
+   the same physical register, the move will be omitted entirely. *)
+type reinterpret_cast =
+  | Int_of_value
+  | Value_of_int
+  | Float_of_float32 (* The high 32 bits of the float are undefined. *)
```

Comment:
(I am slightly confused: aren't we using
128-bits registers, and storing only 32-bits
in memory?)

Reply:
Do you mean that "undefined" should
be interpreted here as "ignored by
the conversion"?

Reply:
Please ensure all this is sufficiently commented (especially in non-target-specific files, so when we come to do e.g. arm64 we remember these points).

Reply:
Yeah, clarified the comment. I think there aren't any other concerns for arm64 here